lib/ConstExtract/ConstExtract.cpp

```diff
@@ -198,24 +198,57 @@ static std::shared_ptr<CompileTimeValue> extractCompileTimeValue(Expr *expr) {
   return std::make_shared<RuntimeValue>();
 }
 
-static std::shared_ptr<CompileTimeValue>
-extractPropertyInitializationValue(VarDecl *propertyDecl) {
-  if (auto binding = propertyDecl->getParentPatternBinding()) {
-    if (auto originalInit = binding->getOriginalInit(0)) {
-      return extractCompileTimeValue(originalInit);
+static std::vector<CustomAttrValue>
+extractCustomAttrValues(VarDecl *propertyDecl) {
+  std::vector<CustomAttrValue> customAttrValues;
+
+  for (auto *propertyWrapper : propertyDecl->getAttachedPropertyWrappers()) {
+    std::vector<FunctionParameter> parameters;
+
+    if (const auto *args = propertyWrapper->getArgs()) {
+      for (auto arg : *args) {
+        const auto label = arg.getLabel().str().str();
+        auto argExpr = arg.getExpr();
+
+        if (auto defaultArgument = dyn_cast<DefaultArgumentExpr>(argExpr)) {
+          auto *decl = defaultArgument->getParamDecl();
+          if (decl->hasDefaultExpr()) {
+            argExpr = decl->getTypeCheckedDefaultExpr();
+          }
+        }
+        parameters.push_back(
+            {label, argExpr->getType(), extractCompileTimeValue(argExpr)});
+      }
+    }
+    customAttrValues.push_back({propertyWrapper->getType(), parameters});
+  }
+
+  return customAttrValues;
+}
+
+static ConstValueTypePropertyInfo
+extractPropertyTypeInfo(VarDecl *propertyDecl) {
```

Comment:
Nit: The `Info` in `ConstValueTypePropertyInfo` refers to the information collected about the `Type`'s `Property`, rather than the `Type` itself. So the name of this method is not a tiny bit misleading. 

I think just a general `extractPropertyInfo` or `extractTypePropertyInfo` would be clearer. What do you think?

Reply:
Makes sense to me. Will update.